boot/zephyr/main.c

```diff
@@ -76,6 +76,53 @@ static void do_boot(struct boot_rsp *rsp)
     __set_MSP(vt->msp);
     ((void (*)(void))vt->reset)();
 }
+
+#elif defined(CONFIG_XTENSA)
+
+static void copy_img_to_SRAM(int slot, unsigned int hdr_offset)
+{
+    const struct flash_area *fap;
+    int area_id;
+    int rc;
+    unsigned char *dst = (unsigned char *)0xBE030000 + hdr_offset;
+
+    BOOT_LOG_INF("Copying image to SRAM");
+
+    area_id = flash_area_id_from_image_slot(slot);
+    rc = flash_area_open(area_id, &fap);
+    if (rc != 0) {
+	BOOT_LOG_ERR("flash_area_open failed with %d\n", rc);
+        goto done;
+    }
+
+    rc = flash_area_read(fap, hdr_offset, dst, 0x30000);
```

Comment:
the same as above - is this a partition size fetch-able form DT?

Reply:
Yes. This is. I will fetch it from DT. thanks for pointing it out.